### PR TITLE
KEYCLOAK-5846

### PR DESCRIPTION
**Pull request by stianst**

### CODE REVIEW

**Comment by stianst** [comment]
@matthewhelmke please review

File: upgrading/topics/rhsso/changes-71.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+== RH-SSO 7.1
+
+The following changes have occurred from RH-SSO 7.0 to RH-SSO 7.1.
+
+=== Realm Keys
+For RH-SSO 7.0 only one set of keys could be associated with a realm. This meant that when changing the keys all current cookies and tokens would be invalidated and all users would have to re-authenticate. For RH-SSO 7.1 support for multiple keys for one realm has been added. At any given time one set of keys is the active keys used for signatures, but there can be multiple keys used to verify signatures. This means that old cookies and tokens can be verified, then refreshed with the new signatures, allowing users to remain authenticated when keys are changed. There are also some changes to how keys are managed through the Admin Console and Admin REST API; for more details see link:{adminguide_link}#realm_keys[Realm Keys] in the _{adminguide_name}_.
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-71.adoc]: For RH-SSO 7.0, only one set of keys could be associated with a realm. This meant that when changing the keys, all current cookies and tokens would be invalidated and all users would have to re-authenticate. For RH-SSO 7.1, support for multiple keys for one realm has been added. At any given time, one set of keys is the active set used for creating signatures, but there can be multiple keys used to verify signatures. This means that old cookies and tokens can be verified, then refreshed with the new signatures, allowing users to remain authenticated when keys are changed. There are also some changes to how keys are managed through the Admin Console and Admin REST API; for more details see link:{adminguide_link}#realm_keys[Realm Keys] in the _{adminguide_name}_.

File: upgrading/topics/rhsso/changes-71.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+To allow seamless key rotation you must remove hard-coded keys from client adapters. The client adapters will automatically retrieve keys from the server as long as the realm key is not specified. Client adapters will also retrieve new keys automatically when keys are rotated.
+
+=== Client Redirect URI Matching
+For RH-SSO 7.0 query parameters are ignored when matching valid redirect URIs for a client. For RH-SSO 7.1 query parameters are no longer ignored. If you need to include query parameters in the redirect URI you must specify the query parameters in the valid redirect URI for the client (for example, \https://hostname/app/login?foo=bar) or use a wildcard (for example, \https://hostname/app/login/*). Fragments are also no longer permitted in Valid Redirect URIs (that is, \https://hostname/app#fragment).
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-71.adoc]: For RH-SSO 7.0, query parameters are ignored when matching valid redirect URIs for a client. For RH-SSO 7.1, query ...

File: upgrading/topics/rhsso/changes-71.adoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+=== Automatically Redirect to Identity Provider
+For RH-SSO 7.1, identity providers cannot be set as the default authentication provider. To automatically redirect to an identity provider for RH-SSO 7.1 you must now configure the identity provider redirector. For more information see link:{adminguide_link}#default_identity_provider[Default Identity Provider] in the _{adminguide_name}_. If you previously had an identity provider with the default authentication provider option set, this value is automatically used as the value for the identity provider redirector when the server is upgraded to RH-SSO 7.1.
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-71.adoc]: To automatically redirect to an identity provider for RH-SSO 7.1, you must now configure the identity provider redirector. For more information, see...

File: upgrading/topics/rhsso/changes-71.adoc [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+=== Admin REST API
+For RH-SSO 7.0 paginated endpoints in the Admin REST API return all results if the maxResults query parameter was not specified. This could cause issues with a temporary high load and requests timing out when a large number of results were returned (for example, users). For RH-SSO 7.1 a maximum of 100 results are returned if a value for maxResults is not specified. You can return all results by specifying maxResults as -1.
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-71.adoc]: For RH-SSO 7.0, paginated endpoints... 

For RH-SSO 7.1, a maximum...

File: upgrading/topics/rhsso/changes-71.adoc [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+=== Server Configuration
+For RH-SSO 7.0 server configuration is split between the keycloak-server.json file and the standalone/domain.xml or domain.xml file. For RH-SSO 7.1 the keycloak-server.json file has been removed and all server configuration is done through the standalone.xml or domain.xml file. The upgrading procedure for RH-SSO 7.1 automatically migrates the server configuration from the keycloak-server.json file to the standalone.xml or domain.xml file.
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-71.adoc]: For RH-SSO 7.0, server configuration...

For RH-SSO 7.1, the keycloak-server.json file...

File: upgrading/topics/rhsso/changes-71.adoc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+=== Key Encryption Algorithm in SAML Assertions
+For RH-SSO 7.1, keys in SAML assertions and documents are now encrypted using the RSA-OAEP encryption scheme. To use encrypted assertions, ensure your service providers support this encryption scheme. In the event that you have service providers that do not support RSA-OAEP, RH-SSO can be configured to use the legacy RSA-v1.5 encryption scheme by starting the server with the system property “keycloak.saml.key_trans.rsa_v1.5” set to true. If you do this  you should upgrade your service providers as soon as possible to be able to revert to the more secure RSA-OAEP encryption scheme.
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-71.adoc]: If you do this, you should...

File: upgrading/topics/rhsso/changes-72.adoc [hunk 7]
```diff
@@ -0,0 +1,34 @@
+== RH-SSO 7.2
+
+The following changes have occurred from RH-SSO 7.1 to RH-SSO 7.2.
+
+=== New Password Hashing algorithms
+
+We've added two new password hashing algorithms (pbkdf2-sha256 and pbkdf2-sha512). New realms will use the pbkdf2-sha256
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-72.adoc]: Change "We've" to "We have"

File: upgrading/topics/rhsso/changes-72.adoc [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+hashing algorithm with 27500 hashing iterations. Since pbkdf2-sha256 is slightly faster than pbkdf2 the iterations was
+increased to 27500 from 20000.
+
+Existing realms are upgraded if the password policy contains the default value for hashing algorithm (not specified) and
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-72.adoc]: ...for the hashing algorithm...

File: upgrading/topics/rhsso/changes-72.adoc [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+iteration (20000). If you have changed the hashing iterations you need to manually change to pbkdf2-sha256 if you'd like
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-72.adoc]: If you have changed the hashing iterations, you need to

File: upgrading/topics/rhsso/changes-72.adoc [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+to use the more secure hashing algorithm.
+
+=== ID Token requires scope=openid
+
+In RH-SSO 7.0 the ID Token was returned regardless if `scope=openid` query parameter was present or not in authorization
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-72.adoc]: In RH-SSO 7.0, the ID Token

File: upgrading/topics/rhsso/changes-72.adoc [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+request. This is incorrect according to the OpenID Connect specification.
+
+In RH-SSO 7.1 we added this query parameter to adapters, but left the old behavior to accommodate migration.
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-72.adoc]: In RH-SSO 7.1, we added

File: upgrading/topics/rhsso/changes-72.adoc [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+In RH-SSO 7.2 this behavior has changed and the `scope=openid` query parameter is now required to mark the request as an
```
Comment (matthewhelmke) [inline upgrading/topics/rhsso/changes-72.adoc]: In RH-SSO 7.2, this behavior

**Review by matthewhelmke — COMMENTED** [review]
Overall, this looks very good. I have noted a few changes, mostly just to add commas.

**Comment by stianst** [comment]
Updated. I think I got all your suggestions correct, but please check. In the future for this amount of suggested edits a PR to my branch would have been simpler as it was quite tricky to apply the changes from comments.

**Comment by matthewhelmke** [comment]
@stianst The changes look good to me. And, noted. I'll learn how to make a PR to someone else's branch; I'm sure it isn't too difficult, I've just never done it.

**Pull request merged**